k8s-intf/src/bolero/support.rs

```diff
@@ -0,0 +1,584 @@
+            prefix += 1;
+            if prefix > largest_prefix {
+                prefix = 0;
+            }
+        }
+        Some(addrs)
+    }
+}
+
+impl ValueGenerator for UniqueV6InterfaceAddressGenerator {
+    type Output = Vec<String>;
+
+    fn generate<D: Driver>(&self, d: &mut D) -> Option<Self::Output> {
+        if self.count == 0 {
+            return Some(vec![]);
+        }
+        // Calculate a mask so that we get a unique prefix for each address
+        let num_prefix_bits = u128::BITS - self.count.next_power_of_two().leading_zeros();
+        let largest_num_addr_bits = 128 - num_prefix_bits;
+        let smallest_mask = num_prefix_bits;
+
+        let largest_prefix = 1_u128.unbounded_shl(num_prefix_bits) - 1;
+        let mut prefix = d.gen_u128(Bound::Excluded(&0), Bound::Included(&largest_prefix))?;
+        let mut addrs = Vec::with_capacity(usize::from(self.count));
+        for _ in 0..self.count {
+            let mask_len = d.gen_u32(Bound::Included(&smallest_mask), Bound::Included(&128))?;
+            let current_num_prefix_bits = 128 - mask_len;
+            let addr_mask = u128::MAX.unbounded_shr(mask_len);
+            // /127 addresses are special case where the first and last address are not broadcast or network addresses
+            #[allow(clippy::bool_to_int_with_if)]
+            let smallest_addr = if current_num_prefix_bits == 0 || mask_len >= 127 {
+                0
+            } else {
+                1
+            };
+            #[allow(clippy::bool_to_int_with_if)]
+            let largest_addr = if current_num_prefix_bits == 0 {
+                0 // The address is all prefix, no address bits
+            } else {
+                addr_mask - (if mask_len >= 127 { 0 } else { 1 })
+            };
+            let addr_data = d.gen_u128(
+                Bound::Included(&smallest_addr),
+                Bound::Included(&largest_addr),
+            )?;
+
+            let addr_as_u128 = prefix.unbounded_shl(largest_num_addr_bits) | addr_data;
+            let addr = Ipv6Addr::from(addr_as_u128);
+            addrs.push(format!("{addr}/{mask_len}"));
+            prefix += 1;
+            if prefix > largest_prefix {
+                prefix = 0;
+            }
+        }
+        Some(addrs)
+    }
+}
+
+pub const ALPHA_NUMERIC_CHARS: &str =
+    "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+pub struct SourceMacAddrString(String);
+impl SourceMacAddrString {
+    #[must_use]
+    pub fn inner(&self) -> &str {
+        &self.0
+    }
+}
+
+impl AsRef<str> for SourceMacAddrString {
+    fn as_ref(&self) -> &str {
+        &self.0
+    }
+}
+// Only generate lower case hex characters for mac addresses
+// because we cannot customize PartialEq for generated types
+// that use this, and we want to be able to compare generated
+// mac addresses with each other without concern for case.
+impl TypeGenerator for SourceMacAddrString {
```

Comment:
I don't really understand this method.

For one thing, I think you want the LSB _of the first octet_ to be set to 0 for unicast.

But more generally, why not just call the type generator for SourceMacAddr and then .to_string it?